components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -1826,97 +1821,67 @@ public int compare(Tag o1, Tag o2) {
+                List<DevPortalAPIInfo> list = searchAPIs.getDevPortalAPIInfoList();
+                List<Object> apiList = new ArrayList<>();
+                for (DevPortalAPIInfo devPortalAPIInfo : list) {
+                    API mappedAPI = APIMapper.INSTANCE.toApi(devPortalAPIInfo);
+                    Set<String> mappedAPITags = mappedAPI.getTags();
+                    for (String tag: mappedAPITags) {
+                        tempTagSet.add(new Tag(tag));
                     }
                 }
-
-
-            } finally {
-                if (isTenantFlowStarted) {
-                    endTenantFlow();
-                }
             }
-
             synchronized (tagCacheMutex) {
                 lastUpdatedTime = System.currentTimeMillis();
                 this.tagSet = tempTagSet;
             }
-
-        } catch (RegistryException e) {
-        	try {
-        		//Before a tenant login to the store or publisher at least one time,
-        		//a registry exception is thrown when the tenant store is accessed in anonymous mode.
-        		//This fix checks whether query resource available in the registry. If not
-        		// give a warn.
-				if (userRegistry != null && !userRegistry.resourceExists(tagsQueryPath)) {
-					log.warn("Failed to retrieve tags query resource at " + tagsQueryPath);
-					return tagSet == null ? Collections.EMPTY_SET : tagSet;
-				}
-			} catch (RegistryException e1) {
-                // Even if we should ignore this exception, we are logging this as a warn log.
-                // The reason is that, this error happens when we try to add some additional logs in an error
-                // scenario and it does not affect the execution path.
-                log.warn("Unable to execute the resource exist method for tags query resource path : " + tagsQueryPath,
-                         e1);
-            }
-            handleException("Failed to get all the tags", e);
-        } catch (UserStoreException e) {
-            handleException("Failed to get all the tags", e);
+        } catch (APIPersistenceException e) {
+            String msg = "Failed to get API tags";
+            throw new APIManagementException(msg, e);
         }
-
         return tagSet;
     }
 
+    @Override
+    public List<APICategory> getAllCategories(String organizationId) throws APIManagementException {
+        Organization org = new Organization(organizationId);
+        String userName = (userNameWithoutChange != null)? userNameWithoutChange: username;
+        String[] roles = APIUtil.getListOfRoles(userName);
+        Map<String, Object> properties = APIUtil.getUserProperties(userName);
+        UserContext userCtx = new UserContext(userNameWithoutChange, org, properties, roles);
+        List<APICategory> categoriesList = new ArrayList<>();
+        try {
+            DevPortalAPISearchResult searchAPIs = apiPersistenceInstance.searchAPIsForDevPortal(org, "",
+                    0, 25, userCtx);
```

Comment:
I think as a temporary solution, checking 25 APIs will not be enough. Decide on a number with discussing with choreo team regarding a max API count per an org for a short term (for max 2 months).

Reply:
Increased to 35 as per choreo team's request https://github.com/wso2/carbon-apimgt/pull/10139/commits/4c07f307be20e8cc8aed25afb3c4c6ac9dcf23aa